salt/modules/file.py

```diff
@@ -1680,44 +1680,19 @@ def _mkstemp_copy(path,
     return temp_file
 
 
-def _starts_till(src, probe, strip_comments=True):
+def _line_match(src, probe):
     '''
-    Returns True if src and probe at least matches at the beginning till some point.
+    Returns True if the line matches the probe
     '''
-    def _strip_comments(txt):
-        '''
-        Strip possible comments.
-        Usually comments are one or two symbols at the beginning of the line, separated with space
-        '''
-        buff = txt.split(" ", 1)
-        return len(buff) == 2 and len(buff[0]) < 2 and buff[1] or txt
-
-    def _to_words(txt):
-        '''
-        Split by words
-        '''
-        return txt and [w for w in txt.strip().split(" ") if w.strip()] or txt
-
-    no_match = -1
+    no_match = 1
     equal = 0
+
     if not src or not probe:
         return no_match
 
-    src = src.rstrip('\n\r')
-    probe = probe.rstrip('\n\r')
-    if src == probe:
+    if probe == src.rstrip('\r\n'):
```

Comment:
Just one question. Can you elaborate why you are not adding `rstrip('\n\r')` to probe as well?

I also added some tests for this PR here https://github.com/jubrad/salt/pull/1  if you can give that a review and merge so we can ensure this does not regress.

Reply:
Thanks, sorry I believe that we've implemented a workaround so there is less pressure on a fix here from us, as such I've completely stopped thinking about it. 

Thanks for adding the tests.

The probe is coming from an argument to the method. I assumed any newlines in it would be intentional thus shouldn't be stripped, although it would probably cause less confusion and better code to just strip both.